deploy/sample-backup-configs/sample_config_build_farm.yaml

```diff
@@ -0,0 +1,24 @@
+### Build Farm ###
+
+# THIS WILL BE USED IF BUILDFARM NOT SPECIFIED IN RECIPE
+# Note: For large designs (ones that would fill a EC2.2xlarge/Xilinx VU9P)
+# Vivado uses in excess of 32 GiB. Keep this in mind when selecting a
+# non-default instance type.
+default-build-farm:
+    build-farm-type: aws-ec2
+    args:
+        instance-type: z1d.2xlarge
+        build-instance-market: ondemand
+        spot-interruption-behavior: terminate
+        spot-max-price: ondemand
+        # change location of build on build host
+        build-dir: null
+
+local-build-farm:
+    build-farm-type: unmanaged
+    args:
+        default-build-dir: null
+        build-farm-hosts:
+            - localhost:
+                # change location of build on build host (abs. path)
+                override-build-dir: null
```

Comment:
You would have to parse the `buildfarmhostdispatcher.py`. Once documentation is added for all this stuff presumably it would point to that file (and that file would have nice comments indicating the args).

Again, the `default-build-dir` and `override-build-dir` being `null` will be documented properly later. I'll update the comments with the args. By default, the build directory on the build host defaults to the home directory (`$HOME/firesim-build`). This allows you to override that (for ex. in the case that you are building locally and want builds to be located in a distinct place).

I'm not entirely sure what your last question is asking. The initial key in the list (i.e. in the example case `localhost`) is the IP address to do the build on. So this is valid yaml:

```yaml
# default to this folder name + "/firesim-build"
default-build-dir: /scratch/abe-gonzalez-folder
build-farm-hosts:
  # use localhost IP address and don't override the default build dir
  - localhost
  # use other IP address (don't override default build dir)
  - "192.168.1.111"
  # use other IP address (override default build dir for just this case)
  - "192.111.2.223":
    override-build-dir: /scratch/specific-build-host-build-dir
```

Reply:
My last question was just asking for what the schema you're proposing would support because I didn't understand why you would pass `null` to both of those.  I think it would be good to have the "unmanaged" build farm example look more like what you put in your comment above because it explains how you expect your 'unmanaged' BuildFarmDispatcher to behave better than what you have in the sample right now.

Only 1 nit,`localhost` isn't an IP address.  It's a hostname that most distros map in `/etc/hosts` to the loopback addresses (as I pointed out elsewhere in your code where you are special casing for localhost).

Reply:
When are you planning to document things?  Before this PR is merged or at some later time?